bindings/profiler.cc

```diff
@@ -140,18 +261,55 @@ Local<Value> TranslateTimeProfile(const CpuProfile* profile) {
 }
 
 NAN_METHOD(StartProfiling) {
-  Local<String> name = info[0].As<String>();
+  if (info.Length() != 2) {
+    return Nan::ThrowTypeError("StartProfling must have two arguments.");
+  }
+  if (!info[0]->IsString()) {
+    return Nan::ThrowTypeError("First argument must be a string.");
+  }
+  if (!info[1]->IsBoolean()) {
+    return Nan::ThrowTypeError("Second argument must be a boolean.");
+  }
 
-  // Sample counts and timestamps are not used, so we do not need to record
-  // samples.
+  Local<String> name =
+      Nan::MaybeLocal<String>(info[0].As<String>()).ToLocalChecked();
+
+// Line-level accurate line information is not available in Node 11 or earlier.
+#if NODE_MODULE_VERSION > NODE_11_0_MODULE_VERSION
+  bool includeLineInfo =
+      Nan::MaybeLocal<Boolean>(info[1].As<Boolean>()).ToLocalChecked()->Value();
+  if (includeLineInfo) {
+    // Sample counts and timestamps are not used, so we do not need to record
+    // samples.
+    cpuProfiler->StartProfiling(name, CpuProfilingMode::kCallerLineNumbers,
+                                false);
+  } else {
+    cpuProfiler->StartProfiling(name, false);
+  }
+#else
   cpuProfiler->StartProfiling(name, false);
+#endif
 }
 
+// stopProfiling(runName: string, includedLineInfo?: boolean): TimeProfile
```

Comment:
nit: it would be good make this stand out as a comment rather than commented out code. It may not survive if it gets mistaken as the latter. 

Something like: 
```
// Signature:
// stopProfiling(runName: string, includedLineInfo?: boolean): TimeProfile
```

Reply:
Done. 
Also made sure all NAN_METHODs had comments like this.